### PR TITLE
Added UPPER_CASE support

### PR DESCRIPTION
**Pull request by vsapronov**
@pinzolo I have added UPPER_CASE support some time ago for my needs. Could you please take a look. If it's possible to merge then I would be able to kill my fork.

### CODE REVIEW

**Comment by pinzolo** [comment]
@vsapronov 
Thanks your suggestion!
I have a question.
Did you have a situation where `strings.ToUpper(casee.ToSnakeCase(s))` didn't work?
Or is it `IsUpperCase` that you need?

File: casee.go [hunk 1]
```diff
@@ -130,6 +130,35 @@ func IsFlatCase(s string) bool {
 	}
 }
 
+// Convert argument to UPPER_CASE style string.
+// If argument is empty, return itself.
+func ToUpperCase(s string) string {
```
Comment (pinzolo) [inline casee.go:135]: I don't think *UpperCase* usually means combined with underscores. So wouldn't `ToUpperSnake` be more appropriate?

File: casee.go [hunk 2]
```diff
@@ -130,6 +130,35 @@ func IsFlatCase(s string) bool {
...
+	if len(s) == 0 {
+		return s
+	}
+
+	fields := splitToLowerFields(s)
+	for i, f := range fields {
+		fields[i] = strings.ToUpper(f)
+	}
+	return strings.Join(fields, "_")
```
Comment (pinzolo) [inline casee.go]: I think that `strings.ToUpper(ToSnakeCase(s))` is simpler implementation.

**Review by pinzolo — COMMENTED** [review]
I left some comments.

File: casee.go (hunk 1 above)
Comment (vsapronov) [reply casee.go:135]: There are different opinions on this. https://stackoverflow.com/questions/17326185/what-are-the-different-kinds-of-cases and https://en.wikipedia.org/wiki/Naming_convention_(programming)#Multiple-word_identifiers
Upper case *without* underscores seems to be consistently called "upper flat case".
I don't care much about name for UPPER_CASE - would take whatever name you would suggest.

File: casee.go (hunk 2 above)
Comment (vsapronov) [reply casee.go]: I can change implementation to this, I think it should work. Though I still think there should be method for UPPER_CASE even if it's so simply implemented - it's just convenient to have all methods in single module for casing conversions.

**Comment by vsapronov** [comment]
@pinzolo Take a look again please. I have changed how `ToUpperCase` is implemented to your suggested one line of code.
For the name, I don't care that much. I've found some sources where people use `UPPER_CASE`. There are  also other variants like `MACRO_CASE` but I don't think "upper case" ever meant for `UPPERCASE` (no underscores) - it seems to be consistently called "flat case" or in this case "upper flat case".

**Comment by pinzolo** [comment]
@vsapronov Thanks your additional works.
I'll merge this.

**Pull request merged**